WEB-INTEGRATION.md

```diff
@@ -0,0 +1,742 @@
+
+The only way JS code can modify the current mapping is through
+`AsyncContext.Variable` and `AsyncContext.Snapshot`’s `run()` methods, which
+switch the context before calling a callback and switch it back after it
+synchronously returns or throws. This ensures that for purely synchronous
+execution, the context is automatically propagated along the data flow. It is
+when tasks and microtasks are queued that web specs must manually track the data
+flow.
+
+The TC39 proposal spec text includes two abstract operations that web specs can
+use to store and switch the context:
+- [`AsyncContextSnapshot()`](https://tc39.es/proposal-async-context/#sec-asynccontextsnapshot)
+  returns the current AsyncContext mapping.
+- [`AsyncContextSwap(context)`](https://tc39.es/proposal-async-context/#sec-asynccontextswap)
+  sets the current AsyncContext mapping to `context`, and returns the previous
+  one. `context` must only be a value returned by one of these two operations.
+
+We propose adding a web spec algorithm “run the AsyncContext Snapshot”, that could be used like this:
+
+> 1. Let _context_ be
+>    [AsyncContextSnapshot](https://tc39.es/proposal-async-context/#sec-asynccontextsnapshot)().
+> 1. [Queue a global task](https://html.spec.whatwg.org/multipage/webappapis.html#queue-a-global-task)
+>    to run the following steps:
+>    1. Run the AsyncContext Snapshot context while performing the following
+>       steps:
+>       1. Perform some algorithm, which might call into JS.
+
+This algorithm, when called with an AsyncContext mapping _context_ and a set of
+steps _steps_, would do the following:
+
+> 1. Let _previousContext_ be
+>    [AsyncContextSwap](https://tc39.es/proposal-async-context/#sec-asynccontextswap)(_context_).
+> 1. Run _steps_. If this throws an exception _e_, then:
+>    1. [AsyncContextSwap](https://tc39.es/proposal-async-context/#sec-asynccontextswap)(_previousContext_).
+>    1. Throw _e_.
+> 1. [AsyncContextSwap](https://tc39.es/proposal-async-context/#sec-asynccontextswap)(_previousContext_).
+
+It might also be possible to integrate “run the AsyncContext Snapshot” together
+with task queueing, since running an AsyncContext Snapshot inside a task will be
+a very common operation.
+
+Since the AsyncContext mapping is a property of the agent, these operations are
+not available in parallel. However, web spec algorithms that run steps in
+parallel before queuing a task that will run JS code must still track the data
+flow along the parallel execution. However, in many cases this will mean
+propagating the snapshotted context along with a callback through the parallel
+algorithms, which will be handled automatically by WebIDL.
+
+AsyncContext mappings also cannot be sent to a different agent. If the data flow
+for some script execution or callback invocation originates from a different
+agent, no context should be restored for that execution. In the case of an
+event, the event would not have a dispatch context.
+
+## Events
```

Comment:
This section seems outdated in light of the above discussion about `dispatchSnapshot` no longer being a thing?